### ISSUE TITLE
[Bug] Automation not working in Firefox after macOS 12 (Monterey) upgrade

### ISSUE THREAD

**Issue by jdmonaco** [darkreader/darkreader#7688]
# Bug Report

## Bug Description

I upgraded my M1 Mac mini to macOS 12+ and updated Firefox to version 95.0, which happened around the same time due to the system upgrade restart. Following that, the DarkReader add-on automation features stopped working, whether based on manual timing or following the system color scheme. This behavior is different from the description in bug #5680: both manually toggling and automatic switching of the system Light/Dark Mode result in no response in the browser/add-on, even after waiting 10-20 seconds. Additionally, activating the "Use system color scheme" option causes DarkReader to switch to browser dark mode regardless of the current system state (light or dark). Unchecking that option returns the browser to the current DarkReader On/Off state of the active website. 

## Website Address

No specific site.

## Steps To Reproduce

- Upgrade a Mac to macOS 12.0 or 12.1 (Monterey)
- Update Firefox to 95.0+
- Either:
  - Set manual times ("active hours") for dark mode in the DarkReader add-on; OR
  - Set "Use system color scheme" in the DarkReader add-on.

## Expected Behavior

- For "active hours" checked: 
  - The add-on should automatically switch to browser dark mode when the system time matches (or passes) the specified start time and deactivate browser dark mode when the system time matches (or passes) the specified end time.
- For "system color scheme" checked: 
  - The add-on should automatically activate/deactivate browser dark mode whenever the macOS dark/light mode is automatically or manually activated; checking the "system" option in the add-on should at that moment activate or deactivate browser dark mode according to the current system state; and unchecking the option should return the browser to the current website's On/Off state within DarkReader.

## Actual Behavior

- For "active hours" checked: 
  - The add-on does not automatically activate/deactivate when the system time passes the start/end times, but the behavior at the moment of checking the "active hours" option is correct according to the current system time.
- For "system color scheme" checked: 
  - The add-on does not activate/deactivate in response to automatic or manual toggling of macOS dark mode; DarkReader activates browser dark mode at the moment of checking the "system" option regardless of the system state; note that unchecking the option results in the expected behavior (return to On/Off state).

## Screenshots

<img width="540" alt="Browser dark mode while system in light mode" src="https://user-images.githubusercontent.com/370169/146680222-6698f886-b3da-416b-b8f5-fa6a67799344.png">

This screenshot was taken immediately after checking the "system" option, which caused the DarkReader add-on to activate browser dark mode despite the system being in light mode (see system info dialog and menubar).

## System Information

- Hardware: Apple Mac mini (M1, 2020)
- OS: macOS 12.0/12.1 (Monterey)
- Browser: Firefox 95 (95.0.1)
- Dark Reader Version: 4.9.43

## Additional Context

The release notes for FF 95 [mentioned a sandboxing change](https://hacks.mozilla.org/2021/12/webassembly-and-back-again-fine-grained-sandboxing-in-firefox-95/) (using `RLBox`), but I don't know whether that might affect how an add-on interacts with system-level notifications like Light/Dark mode changes. 

I have had `privacy.resistFingerprinting = false` in the `about:config`. Enabling or re-disabling that preference makes no difference, nor does restarting the browser.

**Comment by Gusted** [comment]
> * The add-on does not activate/deactivate in response to automatic or manual toggling of macOS dark mode; DarkReader activates browser dark mode at the moment of checking the "system" option regardless of the system state; note that unchecking the option results in the expected behavior (return to On/Off state).

This is a longer issue with firefox whereby they don't give you(limited to the extension) up-to-date results of the system's color scheme.

Interesting about the hours one, we've recently switched to the [alarms web API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/alarms). So they should give you more accurate results, it could be that firefox thinks it's on another timezone, but that wouldn't explain how it does work after manually toggling it. Could you check if firefox correctly has your time, ctrl + shift + i, go to "Console tab" and execute `[new Date().getHours(), new Date().getMinutes()];`, Check if the output [hour, minute] matches with your current time,

Regards,
Gusted

**Comment by jdmonaco** [comment]
Thanks for the response. The main reason I reported the bug is that I experienced this as a regression: the system-based automation was working regularly and quickly (without much of any delay) before I upgraded my system. Due to the simultaneous FF update, I was unsure which upgrade might have caused the issue. Anyway, I guess my system and browser were in some special state where everything worked before I did the upgrades!

I didn't exhaustively test the time-based switching; I mostly just noticed that it hadn't switched some time after it was supposed to have switched. My browser's time is correct. Since the alarms API doesn't persist across sessions, I might have been tricked by my various recent restarts (system and/or browser). I'll give the time switching another chance and see how it goes. 

Anyway, thanks for all your work on this extension! It's great to have it working across all my devices.

Best—
